src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/LocalizedString.shared.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.CommunityToolkit.ObjectModel.Extensions;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+#if !NETSTANDARD1_0
+	public class LocalizedString : ObservableObject
+	{
+		readonly Func<string> generator;
+
+		public LocalizedString(Func<string> generator = null)
+            : this(LocalizationResourceManager.Current, generator)
+        {
+        }
+
+		public LocalizedString(LocalizationResourceManager localizationManager, Func<string> generator = null)
+		{
+			this.generator = generator;
+			localizationManager.WeakSubscribe(this, (t, sender, e) => t.OnPropertyChanged(null));
+		}
+
+		public string Localized => generator?.Invoke();
+
+		[Preserve(Conditional = true)]
+		public static implicit operator LocalizedString(Func<string> func) => new LocalizedString(func);
+	}
```

Comment:
Can we add another implicit operator to convert a `string` into `LocalizedString`? That way we can just type this on VM

`public LocalizedString AppVersion { get; } = string.Format(AppResources.Version, AppInfo.VersionString);`
```suggestion
		[Preserve(Conditional = true)]
		public static implicit operator LocalizedString(Func<string> func) => new LocalizedString(func);
		
		[Preserve(Conditional = true)]
		public static implicit operator LocalizedString(string text) => new LocalizedString(() => text);
	}
```

Reply:
This (with proposed implicit conversion): 
`public LocalizedString AppVersion { get; } = string.Format(AppResources.Version, AppInfo.VersionString);`
Would be the same as this 
`public LocalizedString AppVersion { get; } = () => "Version: 1.0.0";`

If `LocalizedString` would receive already generated string there would be no way to modify it for different culture (string is just as array of `char`s, there is no culture information in it. And if culture changes, it would stay the same array or `char`s). This is why we need a function that can generate string on demand.

Closest thing to what you are proposing is to do implicit conversion from `FormattableString`, but [it's evaluation can be done only once](https://livebook.manning.com/book/c-sharp-in-depth-fourth-edition/chapter-9/point-9410-193-193-0), so it's also not an option (and I don't expect interpolated strings to be used is such way anyway).

Reply:
Yeah, you're right... I forgot that detail. Good catch